File: 5. Longest Palindromic Substring/longest_palindromic_substring.php
<?php
function isPalindrome($sub){
	$lenSub = strlen($sub);
	$mirror = ($lenSub / 2) - (($lenSub / 2) % 1);
	$out = false;
	
	for ($a = 0; $a < $mirror; $a++){
		if ($a == 0){
			$out = ($sub[$a] == $sub[$lenSub - 1 - $a]);
		} else {
			$out = $out && ($sub[$a] == $sub[$lenSub - 1 - $a]);
		}
	}
	
	return ($out);
}

function longestSubString ($text){
	$lenText = strlen($text);
	$longestText = "";
	
	for ($a = 0; $a < $lenText; $a++){
		for ($b = 1; $b < $lenText; $b++){
			$subString = substr($text, $a, ($b + 1));
			
			if (isPalindrome($subString)){
				if (strlen($longestText) < strlen($subString)){
					$longestText = $subString;
				}
			}
		}
	}
	
	return ($longestText);
}

$word = "babad";

echo (longestSubString($word));
echo ("\n\n\n\n");
?>

File: 5. Longest Palindromic Substring/longest_palindromic_substring.rb
def is_palindrome(sub)
	len_sub = sub.length
	mirror = len_sub / 2
	out = false
	
	for a in 0...mirror
		if a == 0
			out = (sub[a] == sub [len_sub - 1 - a])
		else
			out = out and (sub[a] == sub[len_sub - 1 - a])
		end
	end
	
	return out
end

def longest_substring(text)
	len_text = text.length
	longest_text = ""
	
	for a in 0...len_text
		for b in 1...(len_text - a)
			substring = text[a... (a + b + 1)]
			
			if (is_palindrome(substring))
				if (longest_text.length < substring.length)
					longest_text = substring
				
				end
				#break
			end
		end
	end
	
	return longest_text
end

puts(longest_substring("babab"))
puts("\n\n\n\n")
